Bottom: b19e12b5c08786ca73703b6a2928693b93dc12ec
Top:    c2cb5827d5058d1c60153dfc5585a8f00476bbc3
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-01-22 15:06:28 +0800

Refresh of tun-add-missing-xdp-flush

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 0ded28e..a95087b 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1666,13 +1666,13 @@ static struct sk_buff *tun_build_skb(struct tun_struct *tun,
 
 		switch (act) {
 		case XDP_REDIRECT:
-			tfile->xdp_flush_needed = true;
 			get_page(alloc_frag->page);
 			alloc_frag->offset += buflen;
 			err = xdp_do_redirect(tun->dev, &xdp, xdp_prog);
 			if (err)
 				goto err_redirect;
 			rcu_read_unlock();
+			tfile->xdp_flush_needed = true;
 			return NULL;
 		case XDP_TX:
 			xdp_xmit = true;
